dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -2625,7 +2625,7 @@ async fn cmd_db_physical_disks(
 // SERVICES
 
 // Snapshots
-fn format_snapshot(state: &SnapshotState) -> impl Display {
+fn format_snapshot(state: &SnapshotState) -> impl Display + use<> {
```

Comment:
This looks strange! Just making sure I understand: if we didn't have `+ use<>` here, Rust 2024 would infer that the `impl Display` we're returning is tied to the `&SnapshotState` lifetime, but because we say `use<>`, we're explicitly saying that we're returning a type that doesn't?

Which one did Rust 2021 infer?

Reply:
Ah this specific one was added automatically and doesn't end up being strictly required, but in general yeah, in Rust 2021 the output wouldn't be tied to `state`'s lifetime, but in Rust 2024 it is tied to that lifetime by default.